triton-vm/src/table/processor_table.rs

```diff
@@ -1851,432 +2108,467 @@ impl RowPairConstraints {
         self.variables[FULL_WIDTH + usize::from(UniqueClockJumpDifferencesEvalArg)].clone()
     }
 
-    pub fn rpm_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn rpm_next(&self) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(AllClockJumpDifferencesPermArg)].clone()
     }
 
-    pub fn running_evaluation_standard_input_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_evaluation_standard_input_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(InputTableEvalArg)].clone()
     }
-    pub fn running_evaluation_standard_output_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_evaluation_standard_output_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(OutputTableEvalArg)].clone()
     }
-    pub fn running_product_instruction_table_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_product_instruction_table_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(InstructionTablePermArg)].clone()
     }
-    pub fn running_product_op_stack_table_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_product_op_stack_table_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(OpStackTablePermArg)].clone()
     }
-    pub fn running_product_ram_table_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_product_ram_table_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(RamTablePermArg)].clone()
     }
-    pub fn running_product_jump_stack_table_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_product_jump_stack_table_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(JumpStackTablePermArg)].clone()
     }
-    pub fn running_evaluation_to_hash_table_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_evaluation_to_hash_table_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(ToHashTableEvalArg)].clone()
     }
-    pub fn running_evaluation_from_hash_table_next(&self) -> MPolynomial<XFieldElement> {
+    pub fn running_evaluation_from_hash_table_next(
+        &self,
+    ) -> ConstraintCircuitMonad<ProcessorTableChallenges> {
         self.variables[FULL_WIDTH + usize::from(FromHashTableEvalArg)].clone()
     }
 
-    pub fn decompose_arg(&self) -> Vec<MPolynomial<XFieldElement>> {
+    pub fn decompose_arg(&mut self) -> Vec<ConstraintCircuitMonad<ProcessorTableChallenges>> {
```

Comment:
(Why) is this `&mut` necessary?

Reply:
Because the monad's counter is incremented when a new or node is made... OR SO I THOUGHT! Turns out it isn't necessary when mutating values behind reference counters.

Removed that `mut` and many others. Thanks.